profiling/src/collections/string_storage.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2023-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::cell::Cell;
+use std::collections::HashMap;
+use std::hash::BuildHasherDefault;
+use std::ptr;
+use std::rc::Rc;
+
+use super::identifiable::StringId;
+use super::string_table::StringTable;
+
+#[derive(PartialEq, Debug)]
+struct ManagedStringData {
+    str: Rc<str>,
+    last_usage_gen: Cell<Option<u32>>, // TODO: This is not actually being used; maybe remove?
+    cached_seq_num_for: Cell<Option<*const StringTable>>,
+    cached_seq_num: Cell<Option<StringId>>,
```

Comment:
We are storing the cache on each string, but aren't these all added in a batch to the same string table? I think you said that you add all these managed strings to the Profile's string table just before serialization, right? Couldn't we perform a larger batch operation and store the cache there? That way the memory is only used on serialization rather than kept around but largely not being used.

Reply:
As I as looking at this, I did a small tweak to store these as a tuple, rather than as separate entries, as they're related anyway -- 1f2b953cd3fd90df72f6f98a460a682420947b22 .

Your suggestion is interesting, but I'm curious how far were you thinking about the "large batch operation". In particular, were you thinking of moving the cache entirely away from the string table, to the profile? Or even to the caller of the profile?